docs/new-project-setup.md

```diff
@@ -0,0 +1,310 @@
+There are some GitHub actions that will fail because the repository is not yet
+configured to work with the new Snowflake account.
+
+## Deploy project infrastructure using Terraform
+
+We will create two separate deployments of the project infrastructure,
+one for development, and one for production.
+In some places we will refer to project name and owner as `<project>` and `<owner>`, respectively,
+following our [naming conventions](./naming-conventions.md).
+You should substitute the appropriate names there.
+
+### Create the dev configuration
+
+1. In the new git repository, create a directory to hold the development Terraform configuration:
+    ```bash
+    mkdir -p terraform/environments/dev/
+    ```
+    The location of this directory is by convention, and subject to change.
+1. Copy the terraform configuration from
+    [here](https://github.com/cagov/data-infrastructure/blob/main/terraform/snowflake/environments/dev/main.tf)
+    to your `dev` directory.
+1. In the "elt" module of `main.tf`, change the `source` parameter to point to
+    `"github.com/cagov/data-infrastructure.git//terraform/snowflake/modules/elt?ref=<ref>"`
+    where `<ref>` is the short hash of the most recent commit in the `data-infrastructure` repository.
+1. In the `dev` directory, create a new backend configuration file called `<owner>-<project>-dev.tfbackend`.
+    The file will point to the S3 bucket in which we are storing terraform state:
+    ```hcl
+    bucket = "dse-snowflake-dev-terraform-state"
+    dynamodb_table = "dse-snowflake-dev-terraform-state-lock"
+    key = "<owner>-<project>-dev.tfstate"
+    region = "us-west-2"
+    ```
+1. In the `dev` directory, create a terraform variables file called `terraform.tfvars`,
+    and populate the "elt" module variables. These variables may expand in the future,
+    but at the moment they are just the new Snowflake account locator and the environment
+    (in this case `"DEV"`):
+    ```hcl
+    locator     = "<account-locator>"
+    environment = "DEV"
+    ```
+1. Initialize the configuration:
+    ```bash
+    terraform init -backend-config <owner>-<project>-dev.tfbackend
+    ```
+1. Include both Mac and Linux provider binaries in your terraform lock file.
+    This helps mitigate differences between CI environments and ODI Macs:
+    ```bash
+    terraform providers lock -platform=linux_amd64 -platform=darwin_amd64
+    ```
+1. Add your new `main.tf`, `terraform.tfvars`, `<owner>-<project>-dev.tfbackend`,
+    and terraform lock file to the git repository. Do not add the `.terraform/` directory.
+
+### Deploy the dev configuration
+
+1. Ensure that your local environment has environment variables set for `SNOWFLAKE_ACCOUNT`,
+    `SNOWFLAKE_USER`, `SNOWFLAKE_PRIVATE_KEY_PATH`,  and `SNOWFLAKE_PRIVATE_KEY_PASSPHRASE`,
+    and that they are set to your new account, rather than any other accounts.
+1. Run `terraform plan` to see the plan for the resources that will be created.
```

Comment:
plan failed and it is unclear what to do to fix it. are there any basic troubleshooting tips that make sense to be added here?

Reply:
I don't think that troubleshooting steps are in-scope here. This is intended to be project set-up docs for our team for reference when setting up a new project.